src/haddock/modules/topology/topocg/aa2cg.py

```diff
@@ -412,43 +431,47 @@ def determine_ss(structure, skipss, pdbf_path):
     """
     # calculate SS
     for model in structure:
+        tmp_file_name = f"../{output_path}/{pdbf_path.split('/')[-1][:-4]}_tmp.pdb"
+        create_file_with_cryst(pdbf_path, tmp_file_name)
+
         if skipss:
             continue
         else:
             try:
-                dssp = DSSP(model, pdbf_path)
+                p = subprocess.Popen(["dssp", tmp_file_name, "--output-format", "dssp"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+                dssp_raw, errors = p.communicate()
+                dssp_raw = dssp_raw.split('#')[1].split('\n')[:-1]
+                if Path(tmp_file_name).exists(): Path(tmp_file_name).unlink()
             except:  # TODO: think about making this exception more specific
                 # no secondary structure detected for this model
+                if Path(tmp_file_name).exists(): Path(tmp_file_name).unlink()
                 log.warning('SS could not be assigned, assigning code 1 to all residues')
                 continue
+        
+        dssp = {}
+
+        for line in dssp_raw:
+            if line[11:12] in dssp.keys():
+                dssp[line[11:12]].append(line[16:17])
+            else:
+                dssp[line[11:12]] = []
 
         calculated_chains = list(set([e[0] for e in dssp.keys()]))
 
         # Get SS information and translate it:
         # DSSP > MARTINI > HADDOCK
         # this could still be improved
         for chain in model:
-            if chain.id in calculated_chains:
-                # if ss_dic[model]:
-                # get DSSP value for each residue
-                for r in chain:
-                    try:
-                        r.xtra["SS_DSSP"]
-                    except KeyError:
-                        log.warning('No SS definition found for residue: {} {} {:d}'.
-                                        format(chain.id, r.resname, r.id[1]))
-                        r.xtra["SS_DSSP"] = "-"
-                dssp_dic = collections.OrderedDict([(r, r.xtra["SS_DSSP"]) for r in chain])
-                dssp_ss = "".join(dssp_dic.values())
-                _, martini_types = ss_classification(dssp_ss)  # ancestral function, keep it there
-
-                # transform MARTINI > HADDOCK
-                # and add it to the bfactor col
-                for residue, ss in zip(dssp_dic, martini_types):
-                    code = ss_to_code[ss]
-                    # for atom in residue.get_atoms():
-                    for atom in residue.get_atom():
-                        atom.bfactor = code
+            dssp_ss = "".join(dssp[chain.id])
+            _, martini_types = ss_classification(dssp_ss)  # ancestral function, keep it there
+
+            # transform MARTINI > HADDOCK
+            # and add it to the bfactor col
+            for residue, ss in zip(chain, martini_types):
```

Comment:
Just a note here, this `zip` expects that `chain` and `martini_types` are already in the expected order - maybe its good to add an extra check for it, but if it's too complicated to just add a comment saying that `chain` and `martini_types` order must match